src/redux/search/actionCreater.js

```diff
@@ -0,0 +1,41 @@
+import { API_KEY, API_URL } from 'react-native-dotenv';
+
+export const SEARCH_START = 'search_start';
+export const SEARCH_SUCCESS = 'search_success';
+export const SEARCH_FAILED = 'search_failed';
+
+export function searchStart() {
+  return {
+    type: SEARCH_START,
+  };
+}
+
+export function searchSuccess(result) {
+  return {
+    type: SEARCH_SUCCESS,
+    payload: result,
+  };
+}
+
+export function searchFailed(error) {
+  return {
+    type: SEARCH_FAILED,
+    payload: error,
+  };
+}
+
+export function searchData(input) {
+  return (dispatch) => {
+    const url = new URL(`${API_URL}/search/${input}`);
+    url.searchParams.append('token', API_KEY);
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(response.statusText);
```

Comment:
This might not work as expected. Please check the actual api response and throw the error responsively based on status code.